### PR TITLE
Improve the offline support for the ONNX/TFLite export

### PR DESCRIPTION
**Pull request by fxmarty**
Fixes https://github.com/huggingface/optimum/issues/1104

### CODE REVIEW

File: optimum/exporters/tasks.py [hunk 1]
```diff
@@ -1161,11 +1164,26 @@ def determine_framework(
                 for file in filenames
             ]
         else:
-            if not isinstance(model_name_or_path, str):
-                model_name_or_path = str(model_name_or_path)
-            all_files = huggingface_hub.list_repo_files(model_name_or_path, repo_type="model")
-            if subfolder != "":
-                all_files = [file[len(subfolder) + 1 :] for file in all_files if file.startswith(subfolder)]
+            object_id = model_name_or_path.replace("/", "--")
```
Comment (regisss) [inline optimum/exporters/tasks.py:1167]: Not super familiar with the hub cache, are we sure that replacing "/" by "--" always works?

**Review by regisss — APPROVED** [review]
LGTM!

File: optimum/exporters/tasks.py (hunk 1 above)
Comment (fxmarty) [reply optimum/exporters/tasks.py:1167]: In the "new" caching system I guess yes, I just followed https://github.com/huggingface/huggingface_hub/blob/2c598c484687d586a0ad358301364974912e0262/src/huggingface_hub/file_download.py#L1466-L1467

File: optimum/exporters/onnx/__main__.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import argparse
 from pathlib import Path
 
+from requests.exceptions import ConnectionError as RequestsConnectionError
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/__main__.py:20]: Why changing the name?

File: optimum/exporters/tasks.py [hunk 3]
```diff
@@ -1161,11 +1164,26 @@ def determine_framework(
...
+            full_model_path = Path(cache_dir, f"models--{object_id}")
+            if full_model_path.is_dir():  # explore the cache first
+                # Resolve refs (for instance to convert main to the associated commit sha)
+                revision_file = Path(full_model_path, "refs", "main")
+                if revision_file.is_file():
+                    with open(revision_file) as f:
+                        revision = f.read()
+                cached_path = Path(full_model_path, "snapshots", revision, subfolder)
+                all_files = [
+                    os.path.relpath(os.path.join(dirpath, file), cached_path)
```
Comment (michaelbenayoun) [inline optimum/exporters/tasks.py:1177]: nit: We should stick to the pathlib API here instead of mixing pathlib and os.path IMO

**Review by michaelbenayoun — APPROVED** [review]
LGTM

File: optimum/exporters/onnx/__main__.py (hunk 2 above)
Comment (fxmarty) [reply optimum/exporters/onnx/__main__.py:20]: It collides with python native ConnectionError

File: optimum/exporters/tasks.py (hunk 3 above)
Comment (fxmarty) [reply optimum/exporters/tasks.py:1177]: agreed

**Pull request merged**